src/framework/theme/components/icon/icon.component.ts

```diff
@@ -218,14 +218,17 @@ export class NbIconComponent implements NbIconConfig, OnChanges, OnInit {
   }
 
   ngOnChanges() {
-    if (this.iconDef) {
-      this.iconDef = this.renderIcon(this.icon, this.pack, this.options);
-    }
+    this.iconDef = this.renderIcon(this.icon, this.pack, this.options);
```

Comment:
Not sure, but it looks like we don't need the `ngOnInit` hook if we render icon in `ngOnChanges` anyway. What do you think?

Reply:
Keeping `ngOnInit` as it will help those who create components manually (not in templates) as `ngOnChanges` won't be called for dynamic.